droidlet/dashboard/web/src/StateManager.js

```diff
@@ -472,6 +495,30 @@ class StateManager {
         });
       }
     });
+
+    // Update feed state here because it's received first
```

Comment:
what do you mean by it's received first?

Reply:
This is where the first socket event from the backend to the StateManager is processed

Reply:
we might reorder the backend events over time. so can you rewrite the logic without this assumption?

Reply:
Yep will do!

Reply:
@soumith In order to make the logic order-independent, we would need to check if the mask arrays are equal, which could be expensive. Plus, the data is sent over a few times every second, so the current logic will work even if the order is different as long as the data being sent does not change extremely often. I think it's still a good idea to change the logic but wanted to get your input on how to compare the large mask arrays. 

And the reasoning behind the large array comparisons is to see if we need to update the previous state.

Reply:
send an incrementing `tick` integer with every message that is the same across the message in a given timestamp

Reply:
maybe something like `self.count` of the agent?

Reply:
https://github.com/facebookresearch/droidlet/blob/main/agents/core.py#L32

Reply:
The array comparisons turned out to be fast so I just did that. The 'tick' method might be cleaner and more efficient but would take longer to implement. Let me know if you think it's a good idea to prioritize changing the implementation to using ticks.